scripts/process/grid_from_mask.R

```diff
@@ -0,0 +1,28 @@
+#' take an sf object polygon and make a grid out of it
+#' 
+#' @param viz a vizlab object including a \code{cell_size} parameter input
+#' and a polygon or multipolygon \code{clip_poly}
+#' @details
+#' \code{clip_poly} is the polygon or multipolygon that defines the coverage of
+#' the grid that will be constructed. The \code{cell_size} input 
+#' (in units of the \code{clip_poly} projection) determine the resolution of the grid.
+#' The cell grid is clipped to the mask generated by dissolving features of \code{clip_poly} 
+#' into a single feature (e.g., getting rid of state borders)
+process.grid_from_mask <- function(viz){
+  deps <- readDepends(viz)
+  checkRequired(viz, c("cell_size"))
+  
+  clip_poly <- deps[['clip_poly']]
+  crs <- sf::st_crs(clip_poly)
+  cell_size <- viz[["cell_size"]]
+  
+  # this is a simple way to dissolve multiple polygons into a single mask
+  clip_mask <- clip_poly %>% st_buffer(dist = 0) %>% 
+    mutate(group='mask') %>% group_by(group) %>% summarise()
+  cell_grid <- sf::st_make_grid(clip_mask, cellsize = cell_size, crs = crs)
+  cell_grid_clipped <- sf::st_intersection(cell_grid, clip_mask)
+  
+  # add an id to be used for adding data later:
+  cell_grid_clipped$id <- paste('cell_', 1:length(cell_grid_clipped), sep= '')
```

Comment:
I guess this doesn't work and modifies the object in a weird way. plotting works if you skip this line, but if you include it:
```r
plot(cell_grid_clipped, col = 'red')
Error in UseMethod("st_bbox") : 
  no applicable method for 'st_bbox' applied to an object of class "character"
```
and while dplyr verbs work for some `sf` classes, not this one:
```r
cell_grid_clipped <- mutate(cell_grid_clipped, id = 'test')
Error in UseMethod("mutate_") : 
  no applicable method for 'mutate_' applied to an object of class "c('sfc_GEOMETRY', 'sfc')"
```